File: java/7.java
class Solution {
    public int reverse(int x) {

        try {
            int x2 = x < 0 ? -1 * x : x;

            char[] xCharArray = String.valueOf(x2).toCharArray();
            StringBuilder returnString = new StringBuilder();

            for (int i = xCharArray.length - 1; i >= 0; i--) {
                returnString.append(xCharArray[i]);
            }

            int returnInt = Integer.parseInt(returnString.toString());
            returnInt = x < 0 ? -1 * returnInt : returnInt;

            return returnInt;
        } catch(Exception e){
            return 0;
        }
    }
}

public class Main {
    public static void main(String[] args) {
        Solution solution = new Solution();

        int input1 = 120;
        int result1 = solution.reverse(input1);
        System.out.println(result1);
    }
}

File: kotlin/7.kt
import java.lang.NumberFormatException
import kotlin.math.absoluteValue

class Solution {
    fun reverse(x: Int): Int {
        return try {
            val reversedX = x.absoluteValue.toString().reversed().toInt()
            if( x < 0 ){
                reversedX * -1
            } else {
                reversedX
            }
        } catch (e : NumberFormatException){
            0
        }
    }
}

fun main() {
    val solution = Solution()

    val input1 = 123
    val result1 = solution.reverse( input1 )
    println( result1 )

    val input2 = -123
    val result2 = solution.reverse( input2 )
    println( result2 )

    val input3 = 120
    val result3 = solution.reverse( input3 )
    println( result3 )

    val input4 = 1534236469
    val result4 = solution.reverse( input4 )
    println( result4 )

}